pkg/controller/model/grafanaServiceAccount.go

```diff
@@ -23,6 +23,24 @@ func getServiceAccountAnnotations(cr *v1alpha1.Grafana, existing map[string]stri
 	return MergeAnnotations(cr.Spec.ServiceAccount.Annotations, existing)
 }
 
+func mergeImagePullSecrets(requested []v1.LocalObjectReference, existing []v1.LocalObjectReference) []v1.LocalObjectReference {
+	if existing == nil {
+		return requested
```

Comment:
Previously returned nil, this caused the infinite loop bug, now resolved

Reply:
Great find!

Reply:
why the hell dint it show up on kubernetes ? i am not an openshift user........  Anyways good for us since you fixed it. 
@HubertStefanski nice work !! :1st_place_medal:

Reply:
I honestly have no idea, but this isn't the first time I've encountered weird differences between Openshift and Kubernetes, but I guess that just means double the fun with testing for both :smile: